### ISSUE TITLE
Python 3.5 missing VCRUNTIME140.dll

### ISSUE THREAD

**Issue by Ivoz** [ContinuumIO/anaconda-issues#443]
C:\Users\Matthew>conda create -n 35 python=3.5
	Fetching package metadata: ....
	Solving package specifications: .
	Package plan for installation in environment C:\Miniconda3\envs\35:

	The following NEW packages will be INSTALLED:

	    pip:        7.1.2-py35_0
	    python:     3.5.0-0
	    setuptools: 18.1-py35_0
	    wheel:      0.24.0-py35_0

	Proceed ([y]/n)? y

	Linking packages ...
	[      COMPLETE      ]|##################################################| 100%
	#
	# To activate this environment, use:
	# > activate 35
	#

	C:\Users\Matthew>activate 35
	Activating environment "35"...
	
	[35] C:\Users\Matthew>python

Gives

"The program can't start because VCRUNTIME140.dll is missing from your computer. Try reinstalling the program to fix this problem."

Twice.

I know this was some kind of issue in the last Release Candidates, e.g: [Issue 24476](https://bugs.python.org/issue24476)

**Comment by msarahan** [comment]
Thanks for reporting this.  We will have an updated build that includes the VC runtime ASAP.  In the meantime, you can fix this yourself by installing the VC redistributable from Microsoft: https://www.microsoft.com/en-us/download/details.aspx?id=48145

**Comment by msarahan** [comment]
Fixed in Python-3.5.0-1.  Please update, and post back if you have further issues.

**Comment by ccordoba12** [comment]
Closing then. @Ivoz, feel free to reopen this issue if you continue to have this problem.

**Comment by cracker8090** [comment]
https://bugs.python.org/file39742/24476_1.patch
It's can handle with the problem.

**Comment by msarahan** [comment]
These DLLs are included with the msvc_runtime package, which the python package depends on.  They are extracted into the <prefix>\Library\bin folder.  If you are updating an old installation, you may need to add that folder to PATH.  Please let us know if this mechanism is not working for you.

**Comment by meisterchecho** [comment]
@msarahan I'm on windows 7 and I added the Lib folder to my PATH and that fixed my  problem, thank you.

**Comment by waveform80** [comment]
I repeatedly encountered this problem when attempting to use the Anaconda3 2.4.0 64-bit installer on Windows 10 (without any other Python installation present). By "repeatedly" I mean that the error pops up several dozen times during the installation (which ultimately doesn't work properly as all the attempts to run the Python executable during the installation for things like menu icon creation, fail).

A temporary workaround, if you don't mind dealing with an older Python version, is to grab the 2.3.0 installer from the "zipped installers" link (which has old versions available).

**Comment by msarahan** [comment]
Yikes.  One other solution is to install the Microsoft redistributable for VS 2015: https://www.microsoft.com/en-us/download/details.aspx?id=48145

Those are files that are extracted during installation, and should be in PATH during installation, but it appears something is going wrong on your system during installation, and those files aren't found.  The Microsoft redistributable places them system-wide, so you should have no further problems.

**Comment by ahmadia** [comment]
Hey @msarahan - I just tried installing this on a friend's laptop today with the linked (2.4.0) installation of Anaconda.  I'll do the workaround and post back if it fixes things.

**Comment by ahmadia** [comment]
I can't even install the C++ 2015 Redistributable on this laptop (just stuck on initializing), this is an unfortunate red flag for Python 3.5 installations :/

**Comment by ResidentMario** [comment]
I had this same issue today: http://stackoverflow.com/questions/34363125/conda-python-environment-missing-dll-file-on-windows

**Comment by msarahan** [comment]
Answer from Stack Overflow:

This issue is because something is going wrong with the msvc_runtime package. That package should contain these DLLs. Can you please verify the contents of your environment's root folder? These DLLs should live both in there, and in (prefix)/Library/bin.

One possible cause might be the wrong version of msvc_runtime: it should say [vc14] somewhere in conda list output for Python 3.5.x, and [vc10] for 3.4.x

One other cause might be that PATH is wrong after activation, and you're not picking up the DLLs correctly. Are you using cmd.exe or powershell? We don't support powershell yet - only cmd.exe is known to work. If using cmd.exe, can you post the output of echo %PATH% please?

**Comment by WentaoXuUSST** [comment]
i got the same error as i changed the path  to Anaconda ,maybe it is helpful.

**Comment by msarahan** [comment]
@WentaoXuUSST can you please clarify what you did?  If you move Anaconda after installing it, then all kinds of stuff might break.  That is highly inadvisable.  A better approach would be to uninstall Anaconda and reinstall it.  If you have environments that you want to keep, the best way to maintain them would be to store them as exported environments, then recreate them.  More info here: http://conda.pydata.org/docs/using/envs.html#export-the-environment-file

**Comment by bastula** [comment]
I'm getting this error with Python 3.5.1 on a default install of miniconda.

It installs `msvc_runtime: 1.0.1-0` by default.

However, If I install numpy later on in the same env, it will then prompt me to update the runtime:

`The following packages will be UPDATED:`

`    msvc_runtime: 1.0.1-0 --> 1.0.1-vc14_0  [vc14]`

I wonder why it doesn't use the vc14 dlls by default? My workaround was to install then uninstall numpy if I don't require it.

**Comment by msarahan** [comment]
The error seems to come about when the Python version is not explicitly specified when creating environments:

    conda create -n without python numpy

versus

    conda create -n with python=3.5 numpy

The default behavior is to install the same Python version as the root environment, but this is not activating features correctly.  For now, please explicitly specify python version when creating environments.

CC @ilanschnell @kalefranz

**Comment by bastula** [comment]
Thanks for the update, I will remember to specify the Python version for now.

**Comment by traff** [comment]
The problem is reproduced by me on several Windows machines even specifing python version explicitly.

**Comment by wlbksy** [comment]
I'm hit by this on *Win7 64bit*   with
`Miniconda3-latest-Windows-x86_64.exe	34.6M	2015-12-17 14:20:29	aa2d1ed86600375a4d81681e0f4c4f18`

the default is
```
msvc_runtime              1.0.1                    vc14_0  [vc14]
```

however, whenever I want to `conda install numpy`, it says
```
The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    msvc_runtime-1.0.1         |                0         1.1 MB
```
I think this is a big mistake because the latter does not have all the ``*.dll`` that I need,
such as ``vcruntime140.dll``,``api-ms-win-crt-runtime-l1-1-0.dll``

**Comment by chemacortes** [comment]
I suppose that *build 0* is an "empty package" (*transactional, maybe?*). It's recommended to *pinning* the version of `msvc_runtime` to *build vc14* in order to assure the DLLs from `VC+14` become installed. Just create a file named 'pinned' into the folder 'conda-meta' with the line:

```plain
msvc_runtime 1.0.1 vc14_0
```

**Comment by msarahan** [comment]
That is a problem with dependencies and the solver.  There should never be
an msvc_runtime package that does NOT have a vc feature.  Please post the
output of

conda info

Thanks,
Michael

**Comment by mcg1969** [comment]
In theory, the new solver (4.0.0 or later) should be preferring the featured version of the package over the non-featured version. If that is not the case then we need a reproducible set of `conda` commands.

But really, we should be removing the non-featured versions of `msvc_runtime` from the repo altogether. *Or*, we replace them with a package that includes *all* of the runtimes.

**Comment by wlbksy** [comment]
···
$ conda info
Current conda install:

             platform : win-64
        conda version : 4.0.0
  conda-build version : not installed
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : C:\Miniconda3  (writable)
  default environment : C:\Miniconda3
     envs directories : C:\Miniconda3\envs
        package cache : C:\Miniconda3\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
···

and @chemacortes  thanks a lot

**Comment by chemacortes** [comment]
@mcg1969 removed the non-featured version of `msvc_runtime` today, the conda-update has not problems. 

But yesterday, I got an output similar to:

```
C:\> conda search msvc_runtime
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ......
msvc_runtime   1.0.0   vc10_0  defaults [vc10]
               1.0.0   vc14_0  defaults [vc14]
               1.0.0    vc9_0  defaults [vc9]
            .  1.0.1        0  defaults
               1.0.1   vc10_0  defaults [vc10]
            *  1.0.1   vc14_0  defaults [vc14]
               1.0.1    vc9_0  defaults [vc9]
C:\> conda update markdown
The following packages will be UPDATED:

    msvc_runtime: 1.0.1-vc14_0 --> 1.0.1

Proceed ([y]/n)? n
```
It intended update the featured version with the non-featured. Both packages, msvc_runtime and markdown, were installed.

My `conda-info` is:

```
C:\Users\jmcortesa>conda info
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : win-32
        conda version : 4.0.0
  conda-build version : 1.19.0
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : C:\Users\jmcortesa\apps\Anaconda3  (writable)
  default environment : C:\Users\jmcortesa\apps\Anaconda3
     envs directories : C:\Users\jmcortesa\apps\Anaconda3\envs
        package cache : C:\Users\jmcortesa\apps\Anaconda3\pkgs
         channel URLs : https://conda.anaconda.org/IOOS/win-32/
                        https://conda.anaconda.org/IOOS/noarch/
                        https://repo.continuum.io/pkgs/free/win-32/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-32/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : C:\Users\jmcortesa\.condarc
    is foreign system : False
```

**Comment by mcg1969** [comment]
Have we confirmed that 4.0.1 fixes this? (Or perhaps the forthcoming 4.0.2?)

**Comment by agrawalramakant** [comment]
does nt work with 3.5.2 either. the same dll problem
